server/events/project_command_builder_helpers.go

```diff
@@ -0,0 +1,316 @@
+		BaseRepo:             ctx.Pull.BaseRepo,
+		EscapedCommentArgs:   escapeArgs(commentArgs),
+		AutomergeEnabled:     automergeEnabled,
+		ParallelApplyEnabled: parallelApplyEnabled,
+		ParallelPlanEnabled:  parallelPlanEnabled,
+		AutoplanEnabled:      projCfg.AutoplanEnabled,
+		Steps:                steps,
+		HeadRepo:             ctx.HeadRepo,
+		Log:                  ctx.Log,
+		PullMergeable:        ctx.PullMergeable,
+		Pull:                 ctx.Pull,
+		ProjectName:          projCfg.Name,
+		ApplyRequirements:    projCfg.ApplyRequirements,
+		RePlanCmd:            commentBuilder.BuildPlanComment(projCfg.RepoRelDir, projCfg.Workspace, projCfg.Name, commentArgs),
+		RepoRelDir:           projCfg.RepoRelDir,
+		RepoConfigVersion:    projCfg.RepoCfgVersion,
+		TerraformVersion:     projCfg.TerraformVersion,
+		User:                 ctx.User,
+		Verbose:              verbose,
+		Workspace:            projCfg.Workspace,
+		PolicySets:           policySets,
+	}
+}
+
+func escapeArgs(args []string) []string {
+	var escaped []string
+	for _, arg := range args {
+		var escapedArg string
+		for i := range arg {
+			escapedArg += "\\" + string(arg[i])
+		}
+		escaped = append(escaped, escapedArg)
+	}
+	return escaped
+}
+
+// Extracts required_version from Terraform configuration.
+// Returns nil if unable to determine version from configuration.
+func getTfVersion(ctx *CommandContext, absProjDir string) *version.Version {
+	module, diags := tfconfig.LoadModule(absProjDir)
+	if diags.HasErrors() {
+		ctx.Log.Err("trying to detect required version: %s", diags.Error())
+		return nil
+	}
+
+	if len(module.RequiredCore) != 1 {
+		ctx.Log.Info("cannot determine which version to use from terraform configuration, detected %d possibilities.", len(module.RequiredCore))
+		return nil
+	}
+	requiredVersionSetting := module.RequiredCore[0]
+
+	// We allow `= x.y.z`, `=x.y.z` or `x.y.z` where `x`, `y` and `z` are integers.
+	re := regexp.MustCompile(`^=?\s*([^\s]+)\s*$`)
+	matched := re.FindStringSubmatch(requiredVersionSetting)
+	if len(matched) == 0 {
+		ctx.Log.Debug("did not specify exact version in terraform configuration, found %q", requiredVersionSetting)
+		return nil
+	}
+	ctx.Log.Debug("found required_version setting of %q", requiredVersionSetting)
+	version, err := version.NewVersion(matched[1])
+	if err != nil {
+		ctx.Log.Debug(err.Error())
+		return nil
+	}
+
+	ctx.Log.Info("detected module requires version: %q", version.String())
+	return version
+}
+
+// getCfg returns the atlantis.yaml config (if it exists) for this project. If
+// there is no config, then projectCfg and repoCfg will be nil.
+func getCfg(
+	ctx *CommandContext,
+	globalCfg valid.GlobalCfg,
+	projectName string,
+	dir string,
+	workspace string,
+	repoDir string,
+) (projectCfg *valid.Project, repoCfg *valid.RepoCfg, err error) {
+	parserValidator := &yaml.ParserValidator{}
```

Comment:
I think this is kept as a dependency since it's used in [server.go](https://github.com/lyft/atlantis/blob/master/server/server.go#L306).  Idc too much but you can create a type which you can inject `ParserValidator` into that gets injected into both command builders.


ie.

```go

type RepoConfigParser struct {
    yamlParser: yaml.ParserValidator
}

func (p *ProjectConfigParser) parse(ctx....) (*valid.RepoCfg. err error) {

}
...
type DefaultProjectCommandBuilder struct {
    RepoCfgParser: RepoConfigParser
}

func (d *DefaultProjectCommandBuilder) buildProjectCommands(...) {

    repoCfg, err := d.RepoCfgParser.parse(repoDir)
    
    buildProjectCommandCtx(repoCfg, ....)
}


```